model/histogram/float_histogram.go

```diff
@@ -321,24 +381,45 @@ func (h *FloatHistogram) Add(other *FloatHistogram) *FloatHistogram {
 	h.PositiveSpans, h.PositiveBuckets = addBuckets(h.Schema, h.ZeroThreshold, false, hPositiveSpans, hPositiveBuckets, otherPositiveSpans, otherPositiveBuckets)
 	h.NegativeSpans, h.NegativeBuckets = addBuckets(h.Schema, h.ZeroThreshold, false, hNegativeSpans, hNegativeBuckets, otherNegativeSpans, otherNegativeBuckets)
 
-	return h
+	h.CustomBounds = nil
+
+	return h, nil
 }
 
 // Sub works like Add but subtracts the other histogram.
-func (h *FloatHistogram) Sub(other *FloatHistogram) *FloatHistogram {
-	otherZeroCount := h.reconcileZeroBuckets(other)
-	h.ZeroCount -= otherZeroCount
+func (h *FloatHistogram) Sub(other *FloatHistogram) (*FloatHistogram, error) {
+	if h.HasCustomBounds() != other.HasCustomBounds() {
+		return nil, ErrHistogramsIncompatibleSchema
+	}
+	if h.HasCustomBounds() && !floatBucketsMatch(h.CustomBounds, other.CustomBounds) {
```

Comment:
I'd move this check into the if on line 355, because that's where we depend on it and want to make sure the schemas align before calling addBuckets

Reply:
discussed IRL